pinot-common/src/main/java/org/apache/pinot/common/utils/DataSchema.java

```diff
@@ -334,9 +311,64 @@ public DataType toDataType() {
       }
     }
 
+    /**
+     * Converts to the internal representation of the value.
+     * <ul>
+     *   <li>BOOLEAN: boolean -> int</li>
+     *   <li>TIMESTAMP: Timestamp -> long</li>
+     *   <li>BYTES: byte[] -> ByteArray</li>
+     *   <li>BOOLEAN_ARRAY: boolean[] -> int[]</li>
+     *   <li>TIMESTAMP_ARRAY: Timestamp[] -> long[]</li>
+     * </ul>
+     */
+    public Object toInternal(Object value) {
+      switch (this) {
+        case BOOLEAN:
+          return ((boolean) value) ? 1 : 0;
```

Comment:
It is not clear to my why do we use int as internal boolean value

Reply:
Added more comments explaining it. The internal stored type for BOOLEAN is INT (similarly the stored type for TIMESTAMP is LONG), and we always use the stored type throughout the engine